pkg/instancetype/instancetype_test.go

```diff
@@ -1670,6 +1670,42 @@ var _ = Describe("Instancetype and Preferences", func() {
 
 				Expect(*vmi.Spec.Domain.Firmware.Bootloader.EFI.SecureBoot).To(Equal(*preferenceSpec.Firmware.PreferredUseSecureBoot))
 			})
+
+			It("should not overwrite user defined value with PreferredUseBiosSerial - bug #10313", func() {
+				preferenceSpec = &instancetypev1beta1.VirtualMachinePreferenceSpec{
+					Firmware: &instancetypev1beta1.FirmwarePreferences{
+						PreferredUseBios:       pointer.Bool(true),
+						PreferredUseBiosSerial: pointer.Bool(true),
+					},
+				}
+				vmi.Spec.Domain.Firmware = &v1.Firmware{
+					Bootloader: &v1.Bootloader{
+						BIOS: &v1.BIOS{
+							UseSerial: pointer.Bool(false),
+						},
+					},
+				}
+				Expect(instancetypeMethods.ApplyToVmi(field, instancetypeSpec, preferenceSpec, &vmi.Spec)).To(BeEmpty())
+				Expect(*vmi.Spec.Domain.Firmware.Bootloader.BIOS.UseSerial).To(BeFalse())
+			})
+
+			It("should not overwrite user defined value with PreferredUseSecureBoot - bug #10313", func() {
+				preferenceSpec = &instancetypev1beta1.VirtualMachinePreferenceSpec{
+					Firmware: &instancetypev1beta1.FirmwarePreferences{
+						PreferredUseEfi:        pointer.Bool(true),
```

Comment:
Could we add 2 more tests to ensure `PreferredUseEfi` and `PreferredUseBios` do nothing if the VM spec contains a bootloader?